### DISCUSSION TITLE
Sending Laravel logs to stdout and Telescope error

### DISCUSSION THREAD

**Discussion by vitalykarasik** [laravel/framework#52606] [Q&A]
I'm trying to use https://stackoverflow.com/a/46767123/499915 workaround for sending logs from Laravel Scheduler jobs to stdout. In fact, it's just using a symlink like 

`ln -sf /proc/1/fd/1 /var/log/laravel-scheduler.log`

This works nice, but Telescope module doesn't like this symlink, it fails with

`Failed to open stream: No such file or directory {"exception":"[object] (ErrorException(code: 0): file_get_contents(/var/logs/laravel-scheduler.log): Failed to open stream: No such file or directory at /var/www/vendor/laravel/telescope/src/Watchers/ScheduleWatcher.php:68)"}`
Is there a way to eliminate this error?

**Comment by wizzymore** [comment]
Why don't you create a custom channel for the commands:

```php
'commands' => [
    'driver' => 'single',
    'path' => '/var/log/laravel-scheduler.log', // or the normal storage_path('logs/laravel-scheduler.log')
    'level' => env('LOG_LEVEL', 'debug'),
    'replace_placeholders' => true,
],
```

And simply use `Log::channel('commands')->info/debug/etc...` or:
```php
$logger - Log::channel('commands');

$logger->info/debug/etc...('Message');
```

**Reply by vitalykarasik** [reply]
To say the truth, I have no answer. I reviewed a lot a discussion with workarounds for Laravel Scheduler logs in containers, and tried some of these workarounds.
My current configuration works OK, I decides just to filter out these spam logs into my FluentBit log shipper config.

Thank you @wizzymore, 
Vitaly

**Reply by wizzymore** [reply]
From my testing, file_get_contents() reads symlinks correctly. Is the scheduler running inside of the container?

The only reason i would see this failing is because the Telescope job that is running that code is not running in the container that has that /var/log/... file.

**Reply by vitalykarasik** [reply]
Thank you again!

> file_get_contents() reads symlinks correctly.

Yes, symlink should be transparent for any 'normal' linux software. I suspected that Periscope uses some system call which doesn't supported in stdout.
Yes, it runs into container. (My customer is in process of migration Laravel application from Linux to K8S. So everything is running as containers now.)

**Comment by ghost** [comment]
Hi. If you use background shell for your scheduled commands (->runInBackground()) logging to stdout is impossible. see https://github.com/laravel/framework/discussions/50085

**Reply by vitalykarasik** [reply]
Thank you!
I don't use runInBackground, and in general logging to stdout works as supposed. The only issue it's Periscope spam.
